process/smartContract/scrCommon/scProcessHelper_test.go

```diff
@@ -0,0 +1,268 @@
+
+		account, err := sch.CheckSCRBeforeProcessing(nil)
+		require.Nil(t, account)
+		require.ErrorIs(t, err, process.ErrNilSmartContractResult)
+	})
+	t.Run("marshaller err should err", func(t *testing.T) {
+		args := getSCProcessHelperArgs()
+		args.Accounts = &testState.AccountsStub{
+			LoadAccountCalled: func(address []byte) (vmcommon.AccountHandler, error) {
+				return &testState.AccountWrapMock{}, nil
+			},
+		}
+		expectedErr := errors.New("expected")
+		args.Marshalizer = &testscommon.MarshallerStub{
+			MarshalCalled: func(obj interface{}) ([]byte, error) {
+				return nil, expectedErr
+			},
+		}
+		sch, _ := scrCommon.NewSCProcessHelper(args)
+
+		scr := &smartContractResult.SmartContractResult{
+			Nonce:   0,
+			RcvAddr: []byte{1},
+			SndAddr: []byte{1},
+		}
+		account, err := sch.CheckSCRBeforeProcessing(scr)
+		require.Nil(t, account)
+		require.ErrorIs(t, err, expectedErr)
+	})
+	t.Run("load sender error should err", func(t *testing.T) {
+		args := getSCProcessHelperArgs()
+		expectedErr := errors.New("expected")
+		senderAddr := []byte{1}
+		receiverAddr := []byte{2}
+		args.Accounts = &testState.AccountsStub{
+			LoadAccountCalled: func(address []byte) (vmcommon.AccountHandler, error) {
+				if bytes.Equal(address, senderAddr) {
+					return nil, expectedErr
+				}
+				return &testState.AccountWrapMock{}, nil
+			},
+		}
+
+		sch, _ := scrCommon.NewSCProcessHelper(args)
+
+		scr := &smartContractResult.SmartContractResult{
+			Nonce:   0,
+			RcvAddr: receiverAddr,
+			SndAddr: senderAddr,
+		}
+		account, err := sch.CheckSCRBeforeProcessing(scr)
+		require.Nil(t, account)
+		require.ErrorIs(t, err, expectedErr)
+	})
+	t.Run("load receiver error should err", func(t *testing.T) {
+		args := getSCProcessHelperArgs()
+		expectedErr := errors.New("expected")
+		senderAddr := []byte{1}
+		receiverAddr := []byte{2}
+		args.Accounts = &testState.AccountsStub{
+			LoadAccountCalled: func(address []byte) (vmcommon.AccountHandler, error) {
+				if bytes.Equal(address, receiverAddr) {
+					return nil, expectedErr
+				}
+				return &testState.AccountWrapMock{}, nil
+			},
+		}
+
+		sch, _ := scrCommon.NewSCProcessHelper(args)
+
+		scr := &smartContractResult.SmartContractResult{
+			Nonce:   0,
+			RcvAddr: receiverAddr,
+			SndAddr: senderAddr,
+		}
+		account, err := sch.CheckSCRBeforeProcessing(scr)
+		require.Nil(t, account)
+		require.ErrorIs(t, err, expectedErr)
+	})
+	t.Run("not nil SCR should not err", func(t *testing.T) {
+		args := getSCProcessHelperArgs()
+		args.Accounts = &testState.AccountsStub{
+			LoadAccountCalled: func(address []byte) (vmcommon.AccountHandler, error) {
+				return &testState.AccountWrapMock{}, nil
+			},
+		}
+		sch, _ := scrCommon.NewSCProcessHelper(args)
+
+		scr := &smartContractResult.SmartContractResult{
+			Nonce:   0,
+			RcvAddr: []byte{1},
+			SndAddr: []byte{1},
+		}
+		account, err := sch.CheckSCRBeforeProcessing(scr)
+		require.NotNil(t, account)
```

Comment:
Perhaps check some data from this result to be valid?

Reply:
added test for actual values